### DISCUSSION TITLE
Unable to disable parallel execution in vstest.console.exe

### DISCUSSION THREAD

**Discussion by kfrajtak** [xunit/xunit#2167] [Question] [answered]
Hi, 
I am trying to completely disable parallel execution of tests. The reason is that the legacy code base has lots of static code that is globally configured.
What I have tried is
* `[assembly: CollectionBehavior(DisableTestParallelization = true)]`
* `[assembly: Parallelize(Workers = 1, Scope = ExecutionScope.MethodLevel)]` (from `Microsoft.VisualStudio.TestTools.UnitTesting` namespace)
* `xunit.runner.json` as shown at the bottom

But still my tests are executed in parallel (or on multiple threads) - see the numbers in brackets where I am dumping value of `System.Threading.Thread.CurrentThread.ManagedThreadId`:
```
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.1 (32-bit Desktop .NET 4.0.30319.42000)
[xUnit.net 00:00:11.35]   Discovering: TestAssembly (app domain = on [shadow copy], method display = Method, method display options = None)
[xUnit.net 00:00:19.72]   Discovered:  TestAssembly (found 253 test cases)
[xUnit.net 00:00:19.73]   Starting:    TestAssembly (parallel test collections = off, max threads = 1)
[xUnit.net 00:00:37.76]         [9] Info: ShowOnCustomerDisplay 9
[xUnit.net 00:00:37.76]
  ? Test3 [337ms]
  ? Test4 [335ms]
[xUnit.net 00:00:38.05]     Test5 [PASS]
[xUnit.net 00:00:38.05]       Output:
[xUnit.net 00:00:38.05]         [19] Info: ShowOnCustomerDisplay 19
[xUnit.net 00:00:38.05]
[xUnit.net 00:00:38.33]     Test6 [PASS]
[xUnit.net 00:00:38.33]       Output:
[xUnit.net 00:00:38.33]         [20] Info: ShowOnCustomerDisplay 20
```

Content of `xunit.runner.json`
```
{
  "$schema": "https://xunit.net/schema/current/xunit.runner.schema.json",
  "diagnosticMessages": true,
  "methodDisplay": "method",
  "maxParallelThreads": 1,
  "parallelizeAssembly": false,
  "parallelizeTestCollections": false
}
```

**Comment by bradwilson** [comment, marked as answer]
Can you provide a repro project?

**Reply by kfrajtak** [reply]
I am sorry, that is not possible. It is not a big issue right now. I think my assumption was wrong and Xunit runs each test in a thread from the ThreadPool.
